library/manageiq_provider.py

```diff
@@ -150,14 +164,14 @@ def __init__(self, module, url, user, password, verify_ssl, ca_bundle_path):
         self.changed       = False
         self.providers_url = self.api_url + '/providers'
 
-    def required_updates(self, provider_id, endpoints, zone_id):
+    def required_updates(self, provider_id, endpoints, zone_id, region):
         """ Checks whether an update is required for the provider
 
         Returns:
-            Empty Hash (None) - If the hostname and port passed equals the
-                                provider's current values
-            Hash of Changes   - Changes that need to be made if any endpoint is
-                                different than the current values of the
+            Empty Hash (None) - If the hostname, port, zone and region passed equals
```

Comment:
TBH, regarding both questions, I haven't tried it myself.
changing the MIQ zone should work, otherwise, it's a bug.